src/file_system.rs

```diff
@@ -156,29 +158,7 @@ impl FileSystemOs {
     #[inline]
     pub fn read_link(path: &Path) -> io::Result<PathBuf> {
         let path = fs::read_link(path)?;
-        cfg_if! {
-            if #[cfg(windows)] {
-                Ok(Self::strip_windows_prefix(path))
-            } else {
-                Ok(path)
-            }
-        }
-    }
-
-    pub fn strip_windows_prefix<P: AsRef<Path>>(path: P) -> PathBuf {
-        const UNC_PATH_PREFIX: &[u8] = b"\\\\?\\UNC\\";
-        const LONG_PATH_PREFIX: &[u8] = b"\\\\?\\";
-        let path_bytes = path.as_ref().as_os_str().as_encoded_bytes();
-        path_bytes
-            .strip_prefix(UNC_PATH_PREFIX)
-            .or_else(|| path_bytes.strip_prefix(LONG_PATH_PREFIX))
-            .map_or_else(
-                || path.as_ref().to_path_buf(),
-                |p| {
-                    // SAFETY: `as_encoded_bytes` ensures `p` is valid path bytes
-                    unsafe { PathBuf::from(std::ffi::OsStr::from_encoded_bytes_unchecked(p)) }
-                },
-            )
```

Comment:
This function was based on this libuv's implementation.
https://github.com/libuv/libuv/blob/d4ab6fbba4669935a6bc23645372dfe4ac29ab39/src/win/fs.c#L2774-L2784
I noticed that the `UNC_PATH_PREFIX` handling is incorrect here.
It should be replacing `\\?\UNC\` with `\\` instead of completely stripping it.

Could you try if this fix works? At least, it should align with Node's behavior for this part.

Reply:
The updated code is doing the same thing (I think).